grammar.js

```diff
@@ -343,12 +344,58 @@ module.exports = grammar({
     },
 
     // LiteralString
-    string: ($) =>
+    string: $ => choice($._string_literal, $._multiline_string_literal),
+
+    _string_literal: $ => choice(
       seq(
-        field('start', alias($._string_start, 'string_start')),
-        field('content', optional(alias($._string_content, 'string_content'))),
-        field('end', alias($._string_end, 'string_end'))
+        '"',
```

Comment:
Yes, with string content now being a named node, and the start/end bits being capturable anonymous nodes, there's no need for them (and imo they add bloat to the tree output..)

That does remind me that any queries capturing inner string contents will need to be updated to use the node not field now, I can take care of that in nvim-ts injections but I don't know (and really, I don't think) any plugins that would be affected by that

Reply:
Are you still working on this PR? In case you're busy with something else, lemme know if you want me to take over.

Reply:
Yeah sorry just forgot to respond. I don't like this convention at all for several reasons:

The named nodes will consequentially just show a field named the same thing practically with an alias hiding the actual content with the name of the node..? That's really bizarre to me and is very unhelpful in debugging parsers, as well as serving no real purpose. `string_content` is already the same for both single and multiline strings via the alias, so that's a moot point, and modifying the queries should not be a factor to keep using worse practices

Reply:
> The named nodes will consequentially just show a field named the same thing practically with an alias hiding the actual content with the name of the node..?

😅 I don't follow that argument. You're already willing to do that here: `alias($._multiline_string_start, '[[')`. `[[` won't always be the actual content, it can be `[=[`, `[==[` etc. If we're hiding the actual content already, what's the point?

Also, you don't really have to `alias` it to `string_start`, I'm only asking to keep the field name. you can just do `field('start', alias('"', '"'))` 


> string_content is already the same for both single and multiline strings via the alias, so that's a moot point

Sorry, but it's not.

For `local a = "a\0bc"`, the tree looks like this:

![image](https://user-images.githubusercontent.com/8050659/235338086-1cf548e3-6a8b-472a-9729-0829c01951e5.png)

The same string has 2 `string_content` and each `escape_sequence` will keep adding more `string_content`. While block (multiline) string will always have only a single `string_content`.

Reply:
well, distinguishing [[ and [======[ is useless imo but I'd be fine to remove that alias for consistency if you'd like - having string content **is** useful however. 

The latter I can agree with having a single upper level node string_content which encapsulates unnamed nodes and escape sequences

Reply:
Not sure what you did, but looks like the `escape_sequence` is completely gone from the tree now.

Reply:
Anyway, thanks for your contribution! 😃

I think we have all the necessary things to make this work. I'll be making some tweaks to revert some unnecessary changes (e.g. many tests were changes without any good reason) and merge this as soon as possible.

Reply:
> Not sure what you did, but looks like the `escape_sequence` is completely gone from the tree now.

No? See:
![image](https://user-images.githubusercontent.com/29718261/235429739-c1216828-feb3-4816-9ac5-1fa265aa193d.png)

Reply:
> (e.g. many tests were changes without any good reason)

Well, the tests would fail otherwise with the grammar changes :laughing: 

As for renaming stuff, I'd like the named node names to not changed, the unnamed (hidden) ones, feel free to edit

Reply:
> No? See:

Did you install with the latest changes? 🤔

In you screenshot, tehre are named `escape_sequence` node,

![image](https://user-images.githubusercontent.com/8050659/235432578-c87ff0f8-1f3a-41fe-bea1-bb1fbbbfd5d4.png)

but these tests are passing and there are no `escape_sequence` node inside the `string_content` node.

Reply:
(@amaanq I've seen that on another PR, too: the changes to the generated files you push are not the ones I get if I run `tree-sitter generate` locally. Might want to investigate that.)

Reply:
I've resolved this. But found another issue: https://github.com/MunifTanjim/tree-sitter-lua/pull/20#discussion_r1181493390

Reply:
probably because I use ahlincs ts fork, but it shouldn't have any major differences at all - \z isn't a valid escape sequence FYI

Reply:
> it shouldn't have any major differences at all - \z isn't a valid escape sequence FYI

It is tho. You can check the reference manual: https://www.lua.org/manual/5.4/manual.html#3.1

> The escape sequence '\z' skips the following span of whitespace characters, including line breaks; it is particularly useful to break and indent a long literal string into multiple lines without adding the newlines and spaces into the string contents.

Reply:
It's not in Lua 5.1, tho. Maybe the language version this parser targets should be explicitly noted in the README?

Reply:
Hmm... So far, I've only tested it against 5.x syntax.

Reply:
> 5.x

That's not a thing, though -- 5.1, 5.2, 5.3, and 5.4 are basically different language dialects (more like Python 2 vs. Python 3).

Reply:
As far as I can say, the syntax is mostly backward compatible? as in, syntax for 5.1 is valid in 5.4?

Not sure how tree-sitter parsers for language like lua should be. Should there be 4 different parser for `5.1` - `5.4`? 🤔

Reply:
There's also LuaJIT. 😂 `\z` is valid in LuaJIT... that's how I discovered it, when working with codes supposed to run with LuaJIT.

Reply:
and `luau`...

I don't know what the best approach here is; supporting Lua 5.4 hasn't been a problem for us so far.
But I do think documentation is never wrong.

Reply:
well, `luau` has some major changes. Like `if` can be used as expression.

> But I do think documentation is never wrong.

Yep, will add that in README.

Reply:
Luau I can extend and modify it accordingly, but I don't think minute false positives matter so much like Clason said prior